src/EnergyPlus/FluidProperties.cc

```diff
@@ -2442,2470 +2617,18 @@ namespace FluidProperties {
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,       0.0,       0.0,       0.0,      0.0,
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,       0.0,       0.0,       0.0,      0.0,
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 2335000.0, 3280000.0, 3040000.0, 3226000.0};
-        DefaultSteamSuperheatedEnthalpyData(112, _) = {
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,       0.0,       0.0,      0.0,
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,       0.0,       0.0,      0.0,
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,       0.0,       0.0,      0.0,
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,       0.0,       0.0,      0.0,
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 3280000.0, 2821000.0, 3085000.0};
-        DefaultSteamSuperheatedEnthalpyData(113, _) = {
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,       0.0,       0.0,      0.0,
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,       0.0,       0.0,      0.0,
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,       0.0,       0.0,      0.0,
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,       0.0,       0.0,      0.0,
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 3280000.0, 2671000.0, 2998000.0};
-        DefaultSteamSuperheatedEnthalpyData(114, _) = {
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,       0.0,       0.0,      0.0,
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,       0.0,       0.0,      0.0,
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,       0.0,       0.0,      0.0,
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,       0.0,       0.0,      0.0,
-            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 3280000.0, 2512000.0, 2906000.0};
-    }
-
-    void DefaultSteamSuperheatedDensityData_initializer(Array2D<Real64> &DefaultSteamSuperheatedDensityData)
```

Comment:
This array was unused, so I removed it. I am not sure if it should be in there, even as Thermodynamics data, for reference.

Reply:
We're not currently testing anything with the superheated steam data, but I'm not sure we totally want to eliminate it here either. Removing these would mean the user would be required to add the dataset objects for R744 when they want to use it, whereas in the original code _I think_ it would just default to water if they don't add the IDF objects. @mjwitte @EnergyArchmage can either of you confirm? I'm inclined to leave these in, but just now converted to `constexpr` arrays.

Reply:
I concluded this based on my IDE (CLion) showing that the [`DefaultSteamSuperheatedDensityData_initializer`](https://github.com/NREL/EnergyPlus/blob/82391071e9a009712653ee6017118dce65dacf90/src/EnergyPlus/FluidProperties.cc#L3943) function was unused. I searched the whole E+ project directory, but couldn't find a `DefaultSteamSuperheatedDensityData_initializer` function call. Currently, the `DefaultSteamSuperheatedDensityData` array is just [initialized](https://github.com/NREL/EnergyPlus/blob/82391071e9a009712653ee6017118dce65dacf90/src/EnergyPlus/FluidProperties.cc#L424) and [stored to RhoshValues](https://github.com/NREL/EnergyPlus/blob/82391071e9a009712653ee6017118dce65dacf90/src/EnergyPlus/FluidProperties.cc#L636). Unless the initializer function is called from somewhere, there is no way to access these arrays, right?

I'm trying to find where `RhoshValues` is used to find if there is some defect there.

Reply:
@jmythms I believe you're right. I also don't see those initializers used anywhere. The superheated `RefrigData` members are just initialized to the uninitialized default superheated steam arrays. This sets the array size, but not the contents.

Reply:
I also checked the [first commit of E+ accessible on Github](https://github.com/NREL/EnergyPlus/blob/e635eded3dfbbc1c7b5e5d2b78022ce13b855931/src/EnergyPlus/FluidProperties.cc#L2200) to see if the call was dropped off accidentally.  But it is the same there as well, just the declaration and definition of `DefaultSteamSuperheatedDensityData_initializer`, but no function calls... Maybe I should dig into the Fortran stuff? 😬

Reply:
Nah, not worth it - leave the Fortran archeology to others ☠️ 

We just need to decide on what it should do at this point. IMO we have the following options.

1. Remove the code, as you've done. We still have the dataset objects so we're not really losing anything.
2. Make it initialize those arrays by default with the steam data. Then we'd have to go and make sure that all of the models that call for the superheated data reflect this behavior in their inputs and documentation. Ew.

My recommendation is no. 1...

Reply:
Yeah, with this data now defined using `constexpr std::array` there is no runtime cost to initialize it, so there is little downside (other than executable size?) for keeping it in there even if it is not currently used.

Reply:
Yes, I am ok with either option :)